src/lib.rs

```diff
@@ -83,62 +84,27 @@ impl FxHasher {
 impl Hasher for FxHasher {
     #[inline]
     fn write(&mut self, mut bytes: &[u8]) {
-        #[cfg(target_pointer_width = "32")]
-        let read_usize = |bytes: &[u8]| u32::from_ne_bytes(bytes[..4].try_into().unwrap());
-        #[cfg(target_pointer_width = "64")]
-        let read_usize = |bytes: &[u8]| u64::from_ne_bytes(bytes[..8].try_into().unwrap());
-
-        let mut hash = FxHasher { hash: self.hash };
-        assert!(size_of::<usize>() <= 8);
+        let mut hash = *self;
+        assert!(size_of::<usize>() <= size_of::<u64>());
         while bytes.len() >= size_of::<usize>() {
-            hash.add_to_hash(read_usize(bytes) as usize);
-            bytes = &bytes[size_of::<usize>()..];
+            let (usize_bytes, rest_bytes) = bytes.split_at(size_of::<usize>());
+            hash.add_to_hash(usize::from_ne_bytes(usize_bytes.try_into().unwrap()));
+            bytes = rest_bytes;
```

Comment:
maybe add a
```rust
/// Polyfill for <https://doc.rust-lang.org/std/primitive.slice.html#method.take + array conversion
fn take_array<'a, T, const N: usize>(slice: &mut &'a [T]) -> &'a [T; N] {
    let front;
    (front, *slice) = slice.split_at(N);
    front.try_into().unwrap()
}
```
so that we can

```suggestion
            hash.add_to_hash(usize::from_ne_bytes(*take_array(&mut bytes)));
```
?

Reply:
Or maybe even
```rust
fn take_array<'a, T, const N: usize>(slice: &mut &'a [T]) -> Option<&'a [T; N]> {
    if slice.len() < N { return None }

    let front;
    (front, *slice) = slice.split_at(N);
    Some(front.try_into().unwrap())
}
```
so that we can 
```rust
    while let Some(&usize_bytes) = take_array(&mut bytes) {
        hash.add_to_hash(usize::from_ne_bytes(usize_bytes));
    }
```

Reply:
Yes, that's an elegant suggestion. Done.